apps/src/randomPointGenerator.cpp

```diff
@@ -1,26 +1,67 @@
 #include <fstream>
 #include <iostream>
 #include <random>
-
 #include <boost/filesystem.hpp>
+#include <boost/program_options.hpp>
 
 #include "envitools/CSVIO.hpp"
 #include "envitools/EnviUtils.hpp"
 
+namespace fs = boost::filesystem;
+namespace po = boost::program_options;
+namespace et = envitools;
+
 // argv == et_PointGenerator xmin, ymin, xmax, ymax, total points, output path
 int main(int argc, char** argv) {
-    if (argc < 7) {
-        std::cout << " Usage: " << argv[0];
-        std::cout << " [xmin] [ymin] [xmax] [ymax] [total points] [output path]"
-                  << std::endl;
+    fs::path inputPath, outputPath;
+
+    // clang-format off
+    po::options_description options("Options");
+    options.add_options()("help,h", "Show this message")(
+        "input-file,i", po::value<std::string>()->required(),
+        "Input file path of ROIs");
```

Comment:
I don't think you're supposed to put `;` until after the last `)` of the options list. So it would be `add_options()(opt1)(opt2)(opt3);` Somehow they setup their option functions to be chainable.